woocommerce/woocommerce-gutenberg-products-block
# Add new hooks related to payment methods and checkout and remove obsolete.

This pull continues the extraction process from #1780.

The focus of this pull was primarily around updating existing hooks and adding new hooks related to the new contexts and payment method interface. This also implements the changes and removes obsolete hooks. Broken down, this pull:

- restructures the `base/hooks` directory to make things a bit more organized. Primarily moving hooks into their own topical folders (I didn't move existing tests yet, that can be done in subsequent pulls that increase test coverage).
- Adds `useStoreOrder` which is a placeholder hook for orderId and order loading. It will be finished in a follow-up pull.
- Adds `useBillingData` hook which exposes the billing data interface from the `PaymentMethodDataContextProvider`.
- Adds `usePaymentMethodInterface` hook (and related typedefs) which exposes the interface from contexts that are fed to payment methods as props. Also implements this hook in the payment method components.
- Add `useCheckoutSubmit` and `useCheckoutRedirectUrl` which both expose slices of the checkout context.
- removes numerous now obsolete hooks that are no longer needed.

## To test

This impacts cart and checkout blocks so testing simply involves ensuring that:

- checkout block loads in frontend and editor without errors in the console.
- cart block loads in the frontend and existing functionality works as expected without any errors in the console.
- cart block loads in the editor without any errors in the console.
